File: python/program/Prodi.py
# Saya Amida Zulfa Laila NIM 2101147 mengerjakan Latihan Praktikum 4
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka
# saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.

class Prodi():
    
    # Private attributes
    __namaProdi=""
    __kodeProdi=0
    __mhs=[]    #composite dari class Mahasiswa
    __dosen=[]  # composite dari class Dosen
    __course=[] # composite dari class course

    # contructor 
    def __init__(self,mhs, dosen,course,kodeProdi=0,namaProdi=""):
        self.__namaProdi = namaProdi
        self.__kodeProdi = kodeProdi
        self.__mhs.append(mhs)
        self.__dosen.append(dosen)
        self.__course.append(course)

    #Getter dan Setter

    # get NamaProdi
    def getNamaProdi(self):
        return self.__namaProdi
    
    # set NamaProdi
    def setNamaProdi(self, namaProdi):
        self.__namaProdi = namaProdi
        
    # get kodeProdi
    def getKodeProdi(self):
        return self.__kodeProdi
    
    # set kodeProdi
    def setKodeProdi(self, kodeProdi):
        self.__kodeProdi = kodeProdi
        
    # get Mahasiswa list
    def getMhs(self):
        return self.__mhs
    
    # set Mahasiswa list
    def setMhs(self, mhs):
        self.__mhs.append(mhs)
        
    # get Dosen list
    def getDsn(self):
        return self.__dosen
    
    # set Dosen list
    def setDsn(self, dosen):
        
        self.__dosen.append(dosen)
        
    # get Course list
    def getCourses(self):
        return self.__course
    
    # set Course list
    def setCourses(self, course):
        
        self.__course.append(course)
    

File: cpp/program/Prodi.cpp
/* Saya Amida Zulfa Laila NIM 2101147 mengerjakan Latihan Praktikum 4
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka
saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin. */

// import library
#include "Header.hh"

/* contructor */
Prodi::Prodi()
{
    kodeProdi = 0;
    namaProdi = "";
}

// constructor with parameter
Prodi::Prodi(int kodeProdi, string namaProdi, Mahasiswa m, Dosen d, Course c)
{
    this->kodeProdi = kodeProdi;
    this->namaProdi = namaProdi;
    this->m.push_back(m);
    this->d.push_back(d);
    this->c.push_back(c);
}

/* Getter dan Setter*/

// Get kodeProdi
int Prodi::getKodeProdi()
{
    return kodeProdi;
}

// set kodeProdi
void Prodi::setKodeProdi(int kodeProdi)
{
    this->kodeProdi = kodeProdi;
}

// get namaProdi
string Prodi::getNamaProdi()
{
    return namaProdi;
}

// set namaProdi
void Prodi::setNamaProdi(string namaProdi)
{
    this->namaProdi = namaProdi;
}

void Prodi::setMahasiswa(Mahasiswa m)
{
    this->m.push_back(m);
}

list<Mahasiswa> Prodi::getMahasiswa()
{
    return m;
}

void Prodi::setDosen(Dosen d)
{
    this->d.push_back(d);
}

list<Dosen> Prodi::getDosen()
{
    return d;
}

void Prodi::setCourse(Course c)
{
    this->c.push_back(c);
}

list<Course> Prodi::getCourse()
{
    return c;
}
/* Destructor */
Prodi::~Prodi()
{
}
